### PR TITLE
Enable shortcuts config on windows and refactor

### LINKED ISSUE #726: Support automatically adding shortcuts on Windows
**Is your feature request related to a problem? Please describe.**

The settings for adding shortcuts automatically when a game is installed are not present on Windows.

**Describe the solution you'd like**

The settings should include the option, after #724 is closed, this can be enabled (and fixed).

**Additional context**

I tried to include this fix in the PR #725, but I hit some issue on windows where the execAsync callback is not getting executed for the `install` command. Something is wrong with the generated command since the uninstall callback works fine.

I was thinking about moving the creation of the shortcuts to the Library component that it's correctly detecting when a game finishes installing.

### PR DESCRIPTION
**Pull request by arielj**
This PR enables the settings to automatically create desktop and start menu shortcuts on windows.

I moved the trigger of those function to the GlobalState class (the execAsync callback was never triggered on windows).

I'll add some comments on the diff.

Again, I'm not really sure how to add automated tests for this but I accept any advice on how to do that.

Closes #726 

-------------------------------------------------------------
Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: electron/legendary/games.ts [hunk 1]
```diff
@@ -268,7 +268,7 @@ class LegendaryGame extends Game {
   /**
    * Adds a desktop shortcut to $HOME/Desktop and to /usr/share/applications
    * so that the game can be opened from the start menu and the desktop folder.
-   * Both can be disabled with enableDesktopShortcutsOnDesktop and enableDesktopShortcutsOnStartMenu
+   * Both can be disabled with addDesktopShortcuts and addStartMenuShortcuts
```
Comment (arielj) [inline electron/legendary/games.ts:271]: seems to be outdated doc

File: electron/main.ts [hunk 2]
```diff
@@ -682,9 +682,13 @@ ipcMain.handle('openDialog', async (e, args) => {
   return { canceled }
 })
 
-ipcMain.handle('openMessageBox', async (e, args) => {
+const openMessageBox = async (args: Electron.MessageBoxOptions) => {
```
Comment (arielj) [inline electron/main.ts:685]: I extracted the content so I can use it later in this same file

File: src/state/GlobalState.tsx [hunk 3]
```diff
@@ -208,17 +208,22 @@ export class GlobalState extends PureComponent<Props> {
...
+          ipcRenderer.send('addShortcut', appName, false)
+        }
+
+        this.handleFilter(filter)
       }
-      this.refreshLibrary({})
-      return notify([title, 'notify.install.imported'])
+
+      notify([title, t(`notify.install.${notifyKey}`)])
+      return this.refreshLibrary({})
```
Comment (arielj) [inline src/state/GlobalState.tsx:226]: I did some refactor in this function to remove some duplications, I can revert things if needed though

File: electron/main.ts [hunk 4]
```diff
@@ -826,9 +830,19 @@ ipcMain.handle('egsSync', async (event, args) => {
   }
 })
 
-ipcMain.on('addShortcut', async (event, appName) => {
+ipcMain.on('addShortcut', async (event, appName: string, fromMenu: boolean) => {
+  const game = Game.get(appName)
+  game.addShortcuts(fromMenu)
+  openMessageBox({
```
Comment (arielj) [inline electron/main.ts:836]: do you think it's best to show the dialog only when added manually?

**Comment by flavioislima** [comment]
Forgot to review this one.
It's ready right?

**Comment by arielj** [comment]
> Forgot to review this one. It's ready right?

yes! it's ready, should I tag the PR with something or tag somebody for review when it's ready?

**Comment by flavioislima** [comment]
> > Forgot to review this one. It's ready right?
> 
> yes! it's ready, should I tag the PR with something or tag somebody for review when it's ready?

Nice. 
Both would be good. You can tag me and others for review and also put the label `ready-for-review`.
so we can take a look when we find some time :)

File: electron/main.ts (hunk 4 above)
Comment (dragonDScript) [reply electron/main.ts:836]: If enableDesktopShortcuts or the other is enabled, I think you shouldn't show this.

**Review by flavioislima — APPROVED** [review]
Great job 👍

**Pull request merged**